workflow-ui/modal-android/src/main/java/com/squareup/workflow1/ui/modal/ViewTrees.kt

```diff
@@ -0,0 +1,61 @@
+package com.squareup.workflow1.ui.modal
+
+import android.content.Context
+import android.content.ContextWrapper
+import android.view.View
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.lifecycle.ViewTreeViewModelStoreOwner
+import androidx.savedstate.SavedStateRegistry
+import androidx.savedstate.SavedStateRegistryOwner
+import androidx.savedstate.ViewTreeSavedStateRegistryOwner
+import com.squareup.workflow1.ui.WorkflowAndroidXSupport.savedStateRegistryOwnerFromViewTreeOrContext
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+
+/**
+ * "Forwards" any view tree tags from the [anchorView]'s view tree to each of the [rootViews].
+ *
+ * Must only be called when [anchorView] is attached, and should be called every time it's attached
+ * in case any of the owners will resolve to different values.
+ *
+ * Forwarding is best-effort – if no owner is set, none will be forwarded.
+ *
+ * @param anchorView A [View] that is attached and exists in the "main" view tree. E.g. a modal
+ * container view. View tree owners will be read from this view.
+ * @param rootViews A [Sequence] of [View]s that will be shown, but will not be part of the same
+ * view tree as [anchorView]. E.g. the root views for some dialogs that are managed by [anchorView].
+ * View tree owners from [anchorView] will be set on all these views.
+ */
+@OptIn(WorkflowUiExperimentalApi::class)
+internal fun linkViewTreeOwners(
```

Comment:
Part of the reason for pulling this out into a separate file is that I think we will need to add a public, general hook into this mechanism to allow external code to wire up arbitrary connections between view trees. Namely, for Compose specifically, we'll need to make sure to forward the `CompositionReference` that's stored in the view tree to the dialog roots as well. I'm planning to put a hook for that in the `ViewEnvironment` and read it here, at which point we'll probably want to move this tree wiring code to the `core-android` module.

Reply:
I get `anchorView` if we are generalizing for this, but I don't "get" `rootViews`, wouldn't it be more like `subTreeViews`?

Reply:
They are effectively the roots of their view trees though. Sure there's a DecorView and whatever on top, but for all of our intents and purposes, these are the root views.

Reply:
yes. Understood. I still like `subTreeRootViews` but fine either way.

Reply:
They're _not_ subtrees though, that's the whole reason any of this logic needs to exist.